core/src/test/scala/eu/shiftforward/apso/json/JsonFormatBuilderSpec.scala

```diff
@@ -115,5 +117,23 @@ class JsonFormatBuilderSpec extends Specification {
       Test4(Some(3), None).toJson(jw) mustEqual """{ "a": 3 }""".parseJson
       Test4(None, None).toJson(jw) mustEqual """{ }""".parseJson
     }
+
+    "allow defining custom JSON Formats with transformations and custom error handling" in {
```

Comment:
Can you add a test here for when you recover from an error with a value, instead of a custom exception?